app/web/features/badges/BadgesPage.tsx

```diff
@@ -0,0 +1,145 @@
+  display: "flex",
+  gap: theme.spacing(2),
+  alignItems: "start",
+}));
+
+const ParentFlexDiv = styled(FlexDiv)(({ theme }) => ({
+  [theme.breakpoints.down("md")]: {
+    flexDirection: "column",
+    gap: theme.spacing(0),
+  },
+}));
+
+const ContentDiv = styled("div")(({ theme }) => ({
+  padding: theme.spacing(2),
+  alignSelf: "stretch",
+  width: "100%",
+}));
+
+const CenteredDiv = styled(ContentDiv)(({ theme }) => ({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+export interface BadgeUserListProps {
+  badgeId: string;
+}
+
+function BadgeUserList({ badgeId }: BadgeUserListProps) {
+  const { t } = useTranslation([PROFILE]);
+
+  const { badgeUserIds, hasNextPage, isFetchingNextPage, fetchNextPage } =
+    useBadgeUsers(badgeId);
+
+  return (
+    <UsersList
+      userIds={badgeUserIds}
+      endChildren={
+        hasNextPage && (
+          <Button loading={isFetchingNextPage} onClick={() => fetchNextPage()}>
+            Load more
+          </Button>
+        )
+      }
+      emptyListChildren={
+        <Typography variant="body1">{t("profile:badges.no_people")}</Typography>
+      }
+    />
+  );
+}
+
+export interface BadgesPageProps {
+  badgeId?: string;
+}
+
+export default function BadgesPage({ badgeId = undefined }: BadgesPageProps) {
+  const { t } = useTranslation([GLOBAL, PROFILE]);
+  const { badges, isLoading: isBadgesLoading } = useBadges();
+  const theme = useTheme();
+
+  const isBelowMd = useMediaQuery(theme.breakpoints.down("md"));
+
+  return (
+    <>
+      <HtmlMeta title={t("global:nav.badges")} />
+      <PageTitle>{t("profile:badges.title")}</PageTitle>
+      <Typography variant="subtitle1">
+        {t("profile:badges.subtitle")}
+      </Typography>
+      <StyledDivider />
+      <ParentFlexDiv>
+        <BadgeList>
+          {badges &&
+            Object.values(badges).map((badge) => (
+              <BadgeListItem key={badge.id}>
+                <Badge badge={badge} />
+              </BadgeListItem>
+            ))}
+        </BadgeList>
+        <StyledDivider
+          orientation={isBelowMd ? "horizontal" : "vertical"}
+          flexItem
+        />
+        {badgeId ? (
+          <ContentDiv>
+            {isBadgesLoading ? (
+              <CenteredSpinner />
+            ) : badges && badgeId in badges ? (
+              <>
+                <FlexDiv>
+                  <Badge badge={badges[badgeId]} />
+                  <Typography variant="body1">
+                    {badges[badgeId].description}
+                  </Typography>
+                </FlexDiv>
+                <BadgeUserList badgeId={badgeId} />
+              </>
+            ) : (
+              <>Badge not found</>
+            )}
+          </ContentDiv>
+        ) : (
+          <CenteredDiv>
+            <Typography variant="subtitle1">
+              {t("profile:badges.click_on_left")}
+            </Typography>
+          </CenteredDiv>
+        )}
```

Comment:
There are a lot of ternary operators in here. Is there a way to break up this logic to make it more readable?

Reply:
Okay I see what you mean, I think it's okay for now as probably this logic won't get more complicated or expanded on much.